src/utils/DeploymentBuilder.ts

```diff
@@ -1,18 +1,80 @@
+import * as hashing from '@dcl/hashing'
+import { hashV1 } from '@dcl/hashing'
 import {
-  buildEntityAndFile,
   ContentFileHash,
+  Entity,
   EntityContentItemReference,
   EntityId,
   EntityMetadata,
   EntityType,
   EntityVersion,
-  Hashing,
   Pointer,
   Timestamp
 } from 'dcl-catalyst-commons'
 import { AuthChain } from 'dcl-crypto'
 
 export class DeploymentBuilder {
+  /**
+   * Take all the entity's data, build the entity file with it, and calculate its id
+   */
+  static async buildEntityAndFile({
+    version,
+    type,
+    pointers,
+    timestamp,
+    content,
+    metadata
+  }: {
+    /** @deprecated version is nolonger required since ADR51 */
+    version?: EntityVersion
+    type: EntityType
+    pointers: Pointer[]
+    timestamp: Timestamp
+    content?: EntityContentItemReference[]
+    metadata?: EntityMetadata
+  }): Promise<{ entity: Entity; entityFile: Uint8Array }> {
+    // Make sure that there is at least one pointer
+    if (pointers.length === 0) throw new Error(`All entities must have at least one pointer.`)
+
+    if (version === EntityVersion.V2) throw new Error(`V2 is not supported.`)
+
+    const entity = {
+      // default version is V3
+      version: version || EntityVersion.V3,
```

Comment:
If the version is deprecated shouldn't we don't do anything about it? Also, wouldn't the client build the latest version of the entitites?

Reply:
the last version is v3. V4 are not needed as of ADR51.

The migration plan would require stop making the "version" field required on the catalyst but I would implement that change later on to not block the migration to the new hashes